File: src/main/scala/robosky/uplands/block/TallUplandsGrassBlock.scala
package robosky.uplands.block

import net.fabricmc.fabric.api.tag.TagRegistry
import net.minecraft.block.{Block, BlockState, FernBlock}
import net.minecraft.tag.BlockTags
import net.minecraft.util.math.BlockPos
import net.minecraft.world.BlockView

class TallUplandsGrassBlock(val settings: Block.Settings) extends FernBlock(settings){
  override def canPlantOnTop(blockState_1: BlockState, blockView_1: BlockView, blockPos_1: BlockPos): Boolean = blockState_1.matches(UplandsBlockTags.PLANTABLE_ON)
}

File: src/main/java/robosky/uplands/block/TallUplandsGrassBlock.java
package robosky.uplands.block;

import net.minecraft.block.BlockState;
import net.minecraft.block.FernBlock;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.BlockView;

public class TallUplandsGrassBlock extends FernBlock {

    public TallUplandsGrassBlock(Settings settings) {
        super(settings);
    }

    @Override
    protected boolean canPlantOnTop(BlockState blockState, BlockView blockView, BlockPos blockPos) {
        return blockState.matches(UplandsBlockTags.PLANTABLE_ON);
    }
}
